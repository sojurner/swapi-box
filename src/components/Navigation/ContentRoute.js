import React from "react";
import { Switch, Route } from "react-router-dom";

import { CardContainer } from "../CardContainer";

export default ({ planets, people, vehicles, toggleFavorites, favorites }) => {
  const savedFavorites = favorites.map(favorite => favorite.name);
  return (
    <Switch>
      <Route
        exact
        path="/planets"
        render={() => (
          <CardContainer data={planets} toggleFavorites={toggleFavorites} savedFavorites={savedFavorites}/>
        )}
      />
      <Route
        exact
        path="/people"
        render={() => (
          <CardContainer data={people} toggleFavorites={toggleFavorites} savedFavorites={savedFavorites}/>
        )}
      />
      <Route
        exact
        path="/vehicles"
        render={() => (
          <CardContainer data={vehicles} toggleFavorites={toggleFavorites} savedFavorites={savedFavorites}/>
        )}
      />
      <Route
        exact
        path="/favorites"
        render={() => (
          <CardContainer data={favorites} toggleFavorites={toggleFavorites} savedFavorites={savedFavorites}/>
        )}
      />
    </Switch>
  );
};
