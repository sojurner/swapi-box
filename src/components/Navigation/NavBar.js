import React from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";
import "./Navigation.css";

export default ({ getData, pressed, favorites }) => {
  return (
    <div className="navigation_bar">
      <div className="planets__CONTAINER">
        <NavLink
          className={
            pressed === "planets" ? "planets__NAV pressed" : "planets__NAV"
          }
          exact
          to="/planets"
          onClick={getData}
        >
          planets
        </NavLink>
      </div>
      <div className="people__CONTAINER">
        <NavLink
          className={
            pressed === "people" ? "people__NAV pressed" : "people__NAV"
          }
          to="/people"
          onClick={getData}
        >
          people
        </NavLink>
      </div>
      <div className="vehicles__CONTAINER">
        <NavLink
          className={
            pressed === "vehicles" ? "vehicles__NAV pressed" : "vehicles__NAV"
          }
          to="/vehicles"
          onClick={getData}
        >
          vehicles
        </NavLink>
      </div>
      <div className="favorites__CONTAINER">
        <NavLink
          className={
            pressed === "favorites"
              ? "favorites__NAV pressed"
              : "favorites__NAV"
          }
          to="/favorites"
        >
          Favorites: {favorites.length}
        </NavLink>
      </div>
      <div className="next_page__CONTAINER">
        <NavLink
          className={
            pressed === "favorites"
              ? "favorites__NAV pressed"
              : "favorites__NAV"
          }
          to="/favorites"
        />
      </div>
    </div>
  );
};
