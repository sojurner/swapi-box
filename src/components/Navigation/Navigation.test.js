import React from "react";

import { NavBar } from "./NavBar.js";
import { ContentRoute } from "./ContentRoute.js";

describe("NavBar", () => {
  let navWrapper;
  let getData;
  let activeButton;
  let favorites;

  beforeEach(() => {
    getData = jest.fn()
    activeButton = 'people'
    favorites = []
    navWrapper = shallow(
    <NavBar
      getData={getData}
      activeButton={activeButton}
      favorites={favorites}
    />)
  })
  
  it("should match snapshot with data passed", () => {

    expect(navWrapper).toMatchSnapshot();
  });

  it.skip("should match snapshot without data passed", () => {

    expect(navWrapper).toMatchSnapshot();
  });

  it.skip("should match snapshot with activeButton types", () => {

    expect(navWrapper).toMatchSnapshot();
  });
});

describe("ContentRoute", () => {
  let contentWrapper;
  let toggleFavorites;
  let people;
  let planets;
  let vehicles;
  let favorites;
  let handlePage;

  beforeEach(() => {
    toggleFavorites = jest.fn();
    people = []
    planets = []
    vehicles = []
    favorites = []
    handlePage = jest.fn();
    contentWrapper = shallow(
    <ContentRoute
      toggleFavorites={toggleFavorites}
      people={people}
      planets={planets}
      vehicles={vehicles}
      favorites={favorites}
      handlePage={handlePage}
    />)
  })

  it("should match snapshot with data object passed", () => {
    expect(contentWrapper).toMatchSnapshot();
  });

  it.skip("should match snapshot without data object passed", () => {
    
    expect(contentWrapper).toMatchSnapshot();
  });

});