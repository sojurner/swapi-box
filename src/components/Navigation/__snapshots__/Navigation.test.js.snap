// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentRoute should match snapshot with data object passed 1`] = `
<Switch>
  <Route
    exact={true}
    path="/planets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/people"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vehicles"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
</Switch>
`;

exports[`ContentRoute should match snapshot without data object passed 1`] = `
<Switch>
  <Route
    exact={true}
    path="/planets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/people"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vehicles"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
</Switch>
`;

exports[`NavBar should match snapshot when activeButton is favorites 1`] = `
<div
  className="navigation_bar"
>
  <div
    className="planets__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="planets__NAV"
      exact={true}
      name="planets"
      onClick={[Function]}
      to="/planets"
    >
      planets
    </NavLink>
  </div>
  <div
    className="people__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="people__NAV"
      name="people"
      onClick={[Function]}
      to="/people"
    >
      people
    </NavLink>
  </div>
  <div
    className="vehicles__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="vehicles__NAV"
      name="vehicles"
      onClick={[Function]}
      to="/vehicles"
    >
      vehicles
    </NavLink>
  </div>
  <div
    className="favorites__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="favorites__NAV pressed"
      to="/favorites"
    >
      Favorites: 
      0
    </NavLink>
  </div>
</div>
`;

exports[`NavBar should match snapshot when activeButton is people 1`] = `
<div
  className="navigation_bar"
>
  <div
    className="planets__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="planets__NAV"
      exact={true}
      name="planets"
      onClick={[Function]}
      to="/planets"
    >
      planets
    </NavLink>
  </div>
  <div
    className="people__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="people__NAV pressed"
      name="people"
      onClick={[Function]}
      to="/people"
    >
      people
    </NavLink>
  </div>
  <div
    className="vehicles__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="vehicles__NAV"
      name="vehicles"
      onClick={[Function]}
      to="/vehicles"
    >
      vehicles
    </NavLink>
  </div>
  <div
    className="favorites__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="favorites__NAV"
      to="/favorites"
    >
      Favorites: 
      0
    </NavLink>
  </div>
</div>
`;

exports[`NavBar should match snapshot when activeButton is planets 1`] = `
<div
  className="navigation_bar"
>
  <div
    className="planets__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="planets__NAV pressed"
      exact={true}
      name="planets"
      onClick={[Function]}
      to="/planets"
    >
      planets
    </NavLink>
  </div>
  <div
    className="people__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="people__NAV"
      name="people"
      onClick={[Function]}
      to="/people"
    >
      people
    </NavLink>
  </div>
  <div
    className="vehicles__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="vehicles__NAV"
      name="vehicles"
      onClick={[Function]}
      to="/vehicles"
    >
      vehicles
    </NavLink>
  </div>
  <div
    className="favorites__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="favorites__NAV"
      to="/favorites"
    >
      Favorites: 
      0
    </NavLink>
  </div>
</div>
`;

exports[`NavBar should match snapshot when activeButton is vehicles 1`] = `
<div
  className="navigation_bar"
>
  <div
    className="planets__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="planets__NAV"
      exact={true}
      name="planets"
      onClick={[Function]}
      to="/planets"
    >
      planets
    </NavLink>
  </div>
  <div
    className="people__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="people__NAV"
      name="people"
      onClick={[Function]}
      to="/people"
    >
      people
    </NavLink>
  </div>
  <div
    className="vehicles__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="vehicles__NAV pressed"
      name="vehicles"
      onClick={[Function]}
      to="/vehicles"
    >
      vehicles
    </NavLink>
  </div>
  <div
    className="favorites__CONTAINER"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="favorites__NAV"
      to="/favorites"
    >
      Favorites: 
      0
    </NavLink>
  </div>
</div>
`;
